### DISCUSSION TITLE
Grafana Histogram Visualisation with Elasticsearch Histogram field type data

### DISCUSSION THREAD

**Discussion by thechile** [grafana/grafana#58503] [Dashboards, Panels & Visualizations]
I have data that is being pre-aggregated before storing in ES. To achive this i use the mapping `"type" : "histogram"`  and send data like

```
{
  "@timestamp": "2022-11-09T03:17:35.770",
  "my_histogram" : {
      "values" : [0.1, 0.2, 0.3, 0.4, 0.5], 
      "counts" : [3, 7, 23, 12, 6] 
   }
}
```

So the data is bucketed, pre-computed with values and corresponding value counts. 

ref: https://www.elastic.co/guide/en/elasticsearch/reference/current/histogram.html#histogram-ex

What i was hoping is to then use Grafana to visualise this in the Histogram Visualisation, but it seems to ignore the `counts` and only uses the `values`. Here is an example using the above document, it only shows count 1 for each bucket instead of the actual counts.

![image](https://user-images.githubusercontent.com/3511504/200794777-2d505e74-9a04-4d49-bdb1-a8264cb05877.png)

Is this possible just now with Grafana ?

I'm using ES: 8.5.0 and Grafana: 9.2.4.

I found some similar Kibana issues, related to the aggregation.

- https://github.com/elastic/kibana/issues/112390
- https://github.com/elastic/kibana/issues/52426

If i run

```
{
  "size": 0,
  "aggs": {
    "magic": {
      "histogram": {
        "field": "my_histogram",
        "interval": 0.1
      }
    }
  }
}
```

then it gives me the correct output/buckets for Grafana but not sure if it's possible to have Grafana make that query i.e. adjust the `aggs` part of the query. 

Thank you.

**Comment by thechile** [comment]
Hi @leeoniya - i saw you done all the hard work on the Histogram panel. Tagging you in case it's something you could add to the panel TODO list and include support for the Elasticsearch histogram fields type by sending the correct aggregation query. Thanks you.

**Comment by gabor** [comment]
@thechile have you tried using Transforms in the dashboard to modify the data to be compatible with the visualization? for example, there's a transform (Organize fields) that allows you to rename fields.

**Comment by leeoniya** [comment]
also see https://github.com/grafana/grafana/pull/61195 for field naming of pre-aggregated histograms. (xMin and/or xMax)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
